desktop/core/src/desktop/js/parse/sql/generic/jison/sql_valueExpression.jison

```diff
@@ -919,3 +919,31 @@ CaseWhenThenListPartTwo_EDIT
      $$ = { caseTypes: [{ types: ['T'] }] };
    }
  ;
+
+// ------------------  :: CASTS  --------------------
+
+PrimitiveTypeOrError
+ : PrimitiveType
+ | error
+ ;
+
+ValueExpression
+ : ValueExpression '::' PrimitiveType
+  {
+    parser.extractExpressionText($$, $1, $2, $3);
+    $$ = { types: [ $3.toUpperCase() ] }
+  }
+ ;
+
+ValueExpression_EDIT
+ : ValueExpression '::' AnyCursor
+  {
+    parser.suggestKeywords(parser.getTypeKeywords());
+    $$ = { types: [ 'T' ] };
+  }
+ | ValueExpression_EDIT '::' PrimitiveTypeOrError
```

Comment:
Was hoping this clause would fix the issue with invalid types like `my_col::dte`, but it didn't  😢